### PR TITLE
Multiplayer interactions

### PR DESCRIPTION
**Pull request by cmyui**

### CODE REVIEW

File: database/migrations/00004_create_relationships_table.down.sql [hunk 1]
```diff
@@ -1,2 +1,5 @@
+-- SQLBook: Code
+
 DROP TABLE relationships;
+
 DROP TYPE RelationshipEnum;
```
Comment (cmyui) [inline database/migrations/00004_create_relationships_table.down.sql]:
```suggestion
DROP TABLE relationships;
DROP TYPE RelationshipEnum;
```

File: server/repositories/multiplayer_match_ids.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+from aioredlock import LockError
+from aioredlock import Sentinel
+
+from server import clients
+from server import settings
+
+# TODO: move this to global
+lock_manager = Aioredlock(
+    redis_connections=[(settings.REDIS_HOST, settings.REDIS_PORT)],  # type: ignore
+)
```
Comment (cmyui) [inline server/repositories/multiplayer_match_ids.py]: should do this before merge probably

**Review by metal-face — APPROVED** [review]

File: server/repositories/multiplayer_match_ids.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
+from server import clients
+
+
+def make_key() -> str:
+    return "server:match_ids"  # TODO: is this weird?
```
Comment (cmyui) [inline server/repositories/multiplayer_match_ids.py]:
```suggestion
    return "server:last_match_id"
```

**Comment by cmyui** [comment]
d

**Pull request merged**